src/topology/separation.lean

```diff
@@ -667,12 +667,54 @@ end
   `x ≠ y` there exists disjoint open sets around `x` and `y`. This is
   the most widely used of the separation axioms. -/
 @[mk_iff] class t2_space (α : Type u) [topological_space α] : Prop :=
-(t2 : ∀x y, x ≠ y → ∃u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅)
+(t2 : ∀ x y, x ≠ y → ∃ u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅)
 
 lemma t2_separation [t2_space α] {x y : α} (h : x ≠ y) :
-  ∃u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅ :=
+  ∃ u v : set α, is_open u ∧ is_open v ∧ x ∈ u ∧ y ∈ v ∧ u ∩ v = ∅ :=
 t2_space.t2 x y h
 
+lemma t2_separation_finset [t2_space α] : ∀ s : finset α,
+  ∃ f : α → set α, set.pairwise_disjoint ↑s f ∧ ∀ x : s, x.1 ∈ f x ∧ is_open (f x) :=
+begin
+  classical,
+  apply @finset.induction _ (λ s : finset α, ∃ f : α → set α, _) _,
+  { exact ⟨λ _, default, λ x, false.elim, λ x, x.2.elim⟩ },
```

Comment:
```suggestion
lemma t2_separation_finset [t2_space α] (s : finset α) :
  ∃ f : α → set α, set.pairwise_disjoint ↑s f ∧ ∀ x ∈ s, x ∈ f x ∧ is_open (f x) :=
begin
  refine finset.induction_on s (by simp) _,
```
I think this is slightly nicer phrased without the coercion to type (you'll need some easy fixes to the below which I haven't included but I think you should be able to figure that bit out?)